Merge

imiskolee opened the pull request.

ghost commented:
## DeepCode's analysis on [#b37ddd](https://www.deepcode.ai/app/gh/imiskolee/vexflow/79a6c489c4c443a4e1ec244e12beec1939828188/imiskolee/vexflow/b37ddd7a228c69b165c96f4431598b703585fc37/pr/_/%2F/code/?utm_source=gh_review&c=0&w=4&i=3&) found:

- :warning: **4** warnings, :information_source: **3** minor issues. :point_down:
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">The static function checkCollision should not be accessed with this. Use the class instance name instead. Occurrences: <ul><li><a href="https://github.com/imiskolee/vexflow/blob/b37ddd7a228c69b165c96f4431598b703585fc37/src/accidental.ts#L198">accidental.ts:198</a></li> <li><a href="https://github.com/imiskolee/vexflow/blob/b37ddd7a228c69b165c96f4431598b703585fc37/src/accidental.ts#L215">accidental.ts:215</a></li> <li><a href="https://github.com/imiskolee/vexflow/blob/b37ddd7a228c69b165c96f4431598b703585fc37/src/accidental.ts#L221">accidental.ts:221</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/imiskolee/vexflow/79a6c489c4c443a4e1ec244e12beec1939828188/imiskolee/vexflow/b37ddd7a228c69b165c96f4431598b703585fc37/pr/_/%2Fsrc%2Faccidental.ts/javascript%2Fdc_interfile_project%2FStaticAccessThis/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Using isNaN may lead to unexpected results. Consider the more robust Number.isNaN instead. Occurrences: <ul><li><a href="https://github.com/imiskolee/vexflow/blob/b37ddd7a228c69b165c96f4431598b703585fc37/src/timesignature.ts#L30">timesignature.ts:30</a></li> <li><a href="https://github.com/imiskolee/vexflow/blob/b37ddd7a228c69b165c96f4431598b703585fc37/src/boundingboxcomputation.ts#L41">boundingboxcomputation.ts:41</a></li> <li><a href="https://github.com/imiskolee/vexflow/blob/b37ddd7a228c69b165c96f4431598b703585fc37/src/boundingboxcomputation.ts#L42">boundingboxcomputation.ts:42</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/imiskolee/vexflow/79a6c489c4c443a4e1ec244e12beec1939828188/imiskolee/vexflow/b37ddd7a228c69b165c96f4431598b703585fc37/pr/_/%2Fsrc%2Ftimesignature.ts/javascript%2Fdc_interfile_project%2FUseNumberIsNan/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Two subsequent branches of an if statement have duplicate bodies (here and here). This may be caused by a copy-paste error. If this usage is intentional, consider merging the branches to avoid code duplication. Occurrences: <ul><li><a href="https://github.com/imiskolee/vexflow/blob/b37ddd7a228c69b165c96f4431598b703585fc37/src/tickcontext.ts#L222">tickcontext.ts:222</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/imiskolee/vexflow/79a6c489c4c443a4e1ec244e12beec1939828188/imiskolee/vexflow/b37ddd7a228c69b165c96f4431598b703585fc37/pr/_/%2Fsrc%2Ftickcontext.ts/javascript%2Fdc_interfile_project%2FDuplicateIfBody/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/imiskolee/vexflow/79a6c489c4c443a4e1ec244e12beec1939828188/imiskolee/vexflow/b37ddd7a228c69b165c96f4431598b703585fc37/pr/_/%2F/code/?utm_source=gh_review&c=0&w=4&i=3&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=imiskolee)
#### 👉 The DeepCode service and API will be deprecated in August, 2021. [Here](https://www.deepcode.ai/blog/migrate-to-snyk-code) is the information how to migrate. Thank you for using DeepCode 🙏 ❤️ !
If you are using our plugins, you might be interested in their successors: [Snyk's JetBrains plugin](https://snyk.co/udpkq) and [Snyk's VS Code plugin](https://snyk.co/udpkr).

rvilarl committed:
refactor any

rvilarl committed:
glyphProps/glyph refactored

rvilarl committed:
clefnote inherits from glyphnote

rvilarl committed:
leger -> ledger

rvilarl committed:
review comments

rvilarl committed:
move rounding to SVGContext

AaronDavidNewman committed:
Make stavetempo use text formatter and not getBBox

for performance reasons

rvilarl committed:
Merge pull request #1457 from rvilarl/fix/1451

styles refactored to consider groups

rvilarl committed:
Merge pull request #1460 from rvilarl/fix/1458a

move rounding to SVGContext

AaronDavidNewman committed:
Use flag metrics to get the width of the flag.

rvilarl committed:
Merge pull request #1453 from rvilarl/fix/1452

Question: Remove assignment of `fillStyle` and `strokeStyle` in tests?

rvilarl committed:
Merge pull request #1462 from AaronDavidNewman/vex-1459-1

Make stavetempo use text formatter and not getBBox

AaronDavidNewman committed:
Annotation and stavesection use text formatter.

AaronDavidNewman committed:
annotation: get actual y of string not max.

AaronDavidNewman committed:
add default metrics and more TextFormatter test cases

AaronDavidNewman committed:
annotations stack up more tightly, only padding between them.

rvilarl committed:
Merge pull request #1468 from AaronDavidNewman/vex-1459-2

AaronDavidNewman committed:
issue 1469

AaronDavidNewman committed:
Same bug in ChordSymbol

The OP for this issue found the same bug in ChordSymbol.  So I added same fix/same test case.

rvilarl committed:
rename js to ts

rvilarl committed:
single quotes

rvilarl committed:
Merge pull request #1471 from AaronDavidNewman/vex-1459-2

issue 1469 - get top Y for top annotation

rvilarl committed:
Merge pull request #1472 from rvilarl/rename/metrics

Rename/metrics

AaronDavidNewman committed:
fix pitch-sorting

Take smallest Y, which is not always last entry for Tab note.

rvilarl committed:
Merge pull request #1473 from AaronDavidNewman/vex-1459-2

fix pitch-sorting

ronyeh committed:
Add generated docs before releasing the beta.

ronyeh committed:
Release VexFlow 4.1.0-beta.0

ronyeh committed:
Remove build/ after releasing version 4.1.0-beta.0 to npm and GitHub.

rvilarl committed:
custom noteheads refactored

rvilarl committed:
Merge pull request #1477 from rvilarl/fix/1415

custom noteheads refactored

rvilarl committed:
fix rebase

ronyeh committed:
Update demos to include the Leland font.

rvilarl committed:
review comments

rvilarl committed:
Merge pull request #1478 from rvilarl/refactor/noteheads1

refactor any in GlyphProps

ronyeh committed:
Update demo to include Leland.

ronyeh committed:
Update docs/api due to recent PR merges.

ronyeh committed:
Release VexFlow 4.1.0-rc.0

ronyeh committed:
Remove build/ after releasing version 4.1.0-rc.0 to npm and GitHub.

ronyeh committed:
Release VexFlow 4.1.0

ronyeh committed:
Remove build/ after releasing version 4.1.0 to npm and GitHub.

rvilarl committed:
Gonville glyphs generated from remap only SMuFL

rvilarl committed:
fix percussion clef code

rvilarl committed:
fix tempo notes

rvilarl committed:
binary visual regression, no threshold

ronyeh committed:
Allow developers to use https://pnpm.io/ without having to check in the
lock file.

ronyeh committed:
Fix build errors.

rvilarl committed:
review comments

rvilarl committed:
Merge pull request #1482 from rvilarl/fix/clefCodeAndTempoNotes

Fix/clef code and tempo notes

rvilarl committed:
stringnumber metrics

rvilarl committed:
chordsymbol metrics

rvilarl committed:
ornament metrics

ronyeh committed:
Image comparison tool so we can more easily review visual diffs between
current and reference builds.

rvilarl committed:
notehead metrics

rvilarl committed:
tuplet metrics

ronyeh committed:
Show images side by side, or overlaid on top of each other.

ronyeh committed:
Add some comments.

rvilarl committed:
stroke metrics

rvilarl committed:
clef metrics

rvilarl committed:
Merge pull request #1487 from rvilarl/refactor/metricsStep

Refactor FontMetrics: stringnumber

ronyeh committed:
Update image compare tool.

ronyeh committed:
Add test images.

rvilarl committed:
Merge pull request #1484 from ronyeh/compare-images

Easily compare images in the current/ and reference/ folders

ronyeh committed:
Update docs/api/ from recent PR merges.

rvilarl committed:
gonville scaled

ronyeh committed:
Improve compare tool with more UI improvements to help with
generating GIFs for reviewing PRs.

Add query params to the compare tool. For example, open the following in
a local browser:

vexflow/tools/compare/index.html?currentLabel=PR-1480&referenceLabel=4.1.0

ronyeh committed:
Merge pull request #1490 from ronyeh/compare-images

Improve compare tool with more UI improvements to help with generating GIFs for reviewing PRs.

ronyeh committed:
Merge pull request #1480 from rvilarl/font/gonvilleSmufl_1

Gonville SMuFL

ronyeh committed:
Release VexFlow 4.2.0-alpha.0

ronyeh committed:
Remove build/ after releasing version 4.2.0-alpha.0 to npm and GitHub.

rvilarl committed:
glypharticulation support

rvilarl committed:
fermataVeryShort glyph added

rvilarl committed:
gonville fixes (arrows, pizzicato, fermata)

rvilarl committed:
regenerate glyphs

rvilarl committed:
Merge pull request #1497 from rvilarl/glyphsDecimals

regenerate glyphs

rvilarl committed:
glyph code support in Articulations

rvilarl committed:
Merge pull request #1486 from rvilarl/refactor/metrics

Refactor FontMetrics

ronyeh committed:
Merge pull request #1495 from rvilarl/feature/glypharticulation

GlyphArticulation support

ronyeh committed:
Update documents from recent PRs.

ronyeh committed:
Release VexFlow 4.2.0-beta.0

ronyeh committed:
Remove build/ after releasing version 4.2.0-beta.0 to npm and GitHub.

rvilarl committed:
fix scale calculation

rvilarl committed:
set pptr as default

rvilarl committed:
fix glyphs inspector

rvilarl committed:
Merge pull request #1510 from rvilarl/fix/glyphInspector

fix glyphs inspector

rvilarl committed:
fix clef width

mscuthbert committed:
first time signature simplifications

mscuthbert committed:
fix tests

mscuthbert committed:
Merge pull request #1503 from rvilarl/fix/1483

Image generation: set pptr as default

mscuthbert committed:
Merge pull request #1500 from rvilarl/fix/1496

gonville fixes (arrows, pizzicato, fermata)

mscuthbert committed:
Documentation improvements

mscuthbert committed:
use nonnull operator to avoid repetition

mscuthbert committed:
More docs fix plus move sumArray

mscuthbert committed:
Reset expTicksUsed after setSoftmaxFactor

rvilarl committed:
Merge pull request #1514 from 0xfe/doc_improvements

A few doc clarifications

mscuthbert committed:
Add docs, setWidth, one bug fixed

Document everything in timeSignature.ts

Add general documentation for what StaveModifier.placeGlyphOnLine does.

Reset width any time Glyph is changed

Fix one bug in TimeSigGlyph where the `.line` of the TimeSignature was being ignored.

rvilarl committed:
Merge pull request #1513 from 0xfe/timesig_simplifications

Time Signature simplifications

rvilarl committed:
scale calculation tested

rvilarl committed:
Merge pull request #1501 from rvilarl/fix/1126

fix scale calculation

ronyeh committed:
Merge pull request #1511 from rvilarl/fix/1504

Fix/1504 Consider actual clef width.

rvilarl committed:
softmax range in tests changed to 1 .. 15

rvilarl committed:
softmax default to 10

rvilarl committed:
Merge pull request #1519 from rvilarl/fix/1502

ronyeh committed:
Update docs before releasing beta.

ronyeh committed:
Release VexFlow 4.2.0-beta.2

ronyeh committed:
Remove build/ after releasing version 4.2.0-beta.2 to npm and GitHub.

ronyeh committed:
Fix a comment.

ronyeh committed:
Addresses a grace note issue mentioned in the #441 thread.

mscuthbert committed:
Merge pull request #1521 from ronyeh/petzold

Addresses a grace note issue mentioned in the #441 thread.

rvilarl committed:
unify metrics

rvilarl committed:
review comments

rvilarl committed:
point fixed on ornaments and text notes

rvilarl committed:
point fixed on time signatures and clefs

ronyeh committed:
Double the timeout before quitting the puppeteer image generation.
A 2016-era Macbook Pro sometimes doesn't finish generating all the
images in time.

ronyeh committed:
Update docs/api/.

ronyeh committed:
Merge pull request #1531 from rvilarl/fix/1523/bravuraLelandPetaluma

Fix/1523/bravura leland petaluma

rvilarl committed:
review comments

rvilarl committed:
fix hardcoded font size

cclauss committed:
Update GitHub Actions

`CI` is ___always__ defined in GitHub Actions.
* https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

cclauss committed:
Update ci-workflow.yaml

cclauss committed:
Upgrade canvas

https://github.com/Automattic/node-canvas/releases

ronyeh committed:
Merge pull request #1534 from cclauss/patch-1

Update GitHub Actions

rvilarl committed:
review comments

rvilarl committed:
Merge pull request #1532 from rvilarl/fix/1527

point fixed on time signatures and clefs

rvilarl committed:
Merge pull request #1533 from rvilarl/fix/1528

point fixed on ornaments and text notes

rvilarl committed:
point fixed on rests

ronyeh committed:
Merge pull request #1536 from rvilarl/point/rests

point fixed on rests

rvilarl committed:
point fixed on pedals

rvilarl committed:
point fixed on MultiMeasureRest and TextDynamics

rvilarl committed:
point fixed on tremolo and parenthesis

rvilarl committed:
gonville with common metrics

rvilarl committed:
fix notehead bounding box

rvilarl committed:
unison configurable

rvilarl committed:
unison tests added

rvilarl committed:
fix unison tests

rvilarl committed:
remove unused import

rvilarl committed:
shift shorter note

rvilarl committed:
Merge pull request #1545 from rvilarl/feature/unisonConfigurable

unison configurable

rvilarl committed:
Merge pull request #1544 from rvilarl/fix/556

fix notehead bounding box

rvilarl committed:
Merge pull request #1538 from rvilarl/point/MultiMeasureRestDynamics

point fixed on MultiMeasureRest and TextDynamics

rvilarl committed:
GonvilleSmufl generation tools

rvilarl committed:
Merge pull request #1548 from rvilarl/gonvillesmufl

GonvilleSmufl generation tools

rvilarl committed:
gonvillesmufl script updated

rvilarl committed:
Merge pull request #1537 from rvilarl/point/pedals

point fixed on pedals

rvilarl committed:
npm-upgrade

rvilarl committed:
doc update

ronyeh committed:
Merge pull request #1541 from rvilarl/point/gonville

gonville with common metrics

rvilarl committed:
tremolo separation to notehead increased

rvilarl committed:
Merge pull request #1540 from rvilarl/point/tremoloParenthesis

point fixed on tremolo and parenthesis

rvilarl committed:
GonvilleSmufl.woff2

rvilarl committed:
unify chordsymbol scales

rvilarl committed:
clef annotations size refactor

rvilarl committed:
coda point standard

rvilarl committed:
tuplet point refactored

rvilarl committed:
Merge pull request #1553 from rvilarl/point/tupletPoint

tuplet point refactored

rvilarl committed:
Merge pull request #1552 from rvilarl/point/clefAnnotations

clef annotations size refactor

rvilarl committed:
coda reduced in StaveRepetition

rvilarl committed:
remove unused fonts in stack

rvilarl committed:
remove unused custom glyphs

rvilarl committed:
remove custom metrics

rvilarl committed:
replace vexAccidentalMicrotonal2 by accidentalBuyukMucennebSharp

rvilarl committed:
Merge pull request #1556 from rvilarl/refactor/custom1

custom: remove unused glyphs, unused fonts in stack and custom metrics

rvilarl committed:
gonville glyph updated

rvilarl committed:
Merge pull request #1557 from rvilarl/refactor/custom2

custom: replace vexAccidentalMicrotonal2 by accidentalBuyukMucennebSharp

rvilarl committed:
use qunit installed library

rvilarl committed:
fix current images generation

rvilarl committed:
use assert in qunit calls

rvilarl committed:
Merge pull request #1564 from rvilarl/update/qunit1

use assert in qunit calls

rvilarl committed:
module name fixed

rvilarl committed:
npm-upgrade

rvilarl committed:
Merge pull request #1562 from rvilarl/update/qunit

Upgrade QUnit and node packages.

rvilarl committed:
doc update

ronyeh committed:
Release VexFlow 4.2.0-beta.3

rvilarl committed:
new fonts

rvilarl committed:
review comments

rvilarl committed:
finale fonts included

ronyeh committed:
Merge pull request #1566 from rvilarl/newfonts

New SMuFL fonts.

ronyeh committed:
Rename LICENSE file.

ronyeh committed:
Bump vexflow-fonts version.

rvilarl committed:
access to font stack

ronyeh committed:
Merge pull request #1567 from rvilarl/fontStackAccess

access to font stack

ronyeh committed:
Merge pull request #1554 from rvilarl/point/coda

Coda

ronyeh committed:
Remove build/ files.

AaronDavidNewman committed:
re-introduce scale to chord symbol metrics

AaronDavidNewman committed:
remove scale parameter, but keep the others for chordsymbol

AaronDavidNewman committed:
remove scale from common metrics also

rvilarl committed:
Merge pull request #1569 from 0xfe/chord-symbol-metrics

re-introduce font-specific formatting to chord symbol metrics

saitonakamura committed:
feat(CanvasContext): Added optional devicePixelRatio argument to resize

It's mainly useful in context of rendering in web worker,
because web worker doesn't have access to devicePixelRatio,
so resize code will be split between
1. main thread asking for a render, providing devicePixelRatio
2. Worker rendering and resizing CanvasContext w/ provided ratio
3. Worker giving back main thread the resize target
4. Main thread applying css width/height provided by the worker

ronyeh committed:
Merge pull request #1572 from saitonakamura/feat-device-pixel-ratio-override

feat(CanvasContext): Added optional devicePixelRatio argument to resize

ronyeh committed:
tests/support/ no longer exists as of PR https://github.com/0xfe/vexflow/pull/1562

ronyeh committed:
Local copy of QUnit 2.19.4.

ronyeh committed:
Fix QUnit dropdown bugs.

ronyeh committed:
Add a top margin.

rvilarl committed:
Merge pull request #1573 from ronyeh/qunit-1

Qunit Fixes/Improvements

rvilarl committed:
fix test scroll

ronyeh committed:
Merge pull request #1574 from rvilarl/qunitTestScroll

fix test scroll

ronyeh committed:
API docs.

ronyeh committed:
Release VexFlow 4.2.0-rc.0

ronyeh committed:
Remove build/ after releasing version 4.2.0-rc.0 to npm and GitHub.

ronyeh committed:
npm audit fix

ronyeh committed:
Fix QUnit bugs.

ronyeh committed:
Release VexFlow 4.2.1

ronyeh committed:
Remove build/ after releasing version 4.2.1 to npm and GitHub.

ronyeh committed:
eslint autofix.

gristow committed:
fix(accidental): extra space in 2nds w/ stems down

gristow committed:
chore(accidental): Remove console.log

gristow committed:
fix(stringnumber): avoid crashes with other left modifiers

gristow committed:
chore: Remove wip testing file

gristow committed:
test(stringnumber): Add test for displaced noteheads/multiple modifiers

gristow committed:
chore(accidentals): Remove FIXME label for things that are fixed.

gristow committed:
chore(accidental): Remove unused, commented-out code

gristow committed:
chore(accidental): Remove unused variable

rvilarl committed:
Merge pull request #1580 from gristow/two-note-chord-spacing

fix: extra space before seconds with modifiers

ronyeh committed:
Add more NPM scripts to expose some grunt tasks.

ronyeh committed:
Merge pull request #1581 from ronyeh/npm-scripts

Add more NPM scripts to expose some grunt tasks.

ronyeh committed:
Release VexFlow 4.2.2

ronyeh committed:
Remove build/ after releasing version 4.2.2 to npm and GitHub.

ronyeh committed:
Add a note about migrating to the VexFlow organization.

ronyeh committed:
Merge pull request #1582 from ronyeh/vexflow5-info

Add a note about migrating to the VexFlow organization.

ronyeh committed:
Update README.md

Update script URL and add information about VexFlow 5.

ronyeh committed:
The grunt get:releases:X.Y.Z script fails because npm pack does not create the directory for you if it doesn't already exist.

This patch allows for easier visual diffing between our current work and a release that has been published on NPM.

This was previously applied to the VexFlow 5+ repo: https://github.com/vexflow/vexflow/pull/8

ronyeh committed:
Merge pull request #1583 from ronyeh/get-releases

FIX: The grunt get:releases:X.Y.Z script fails because `npm pack` does not create the directory for you if it doesn't already exist. This patch allows for easier visual diffing between our current work and a release that has already been published on NPM.

accelerationa committed:
return noteheads after build

ronyeh committed:
Merge pull request #1590 from accelerationa/yluo/return-note-heads-after-build

return noteheads after build

ronyeh committed:
Release VexFlow 4.2.3

ronyeh committed:
Remove build/ after releasing version 4.2.3 to npm and GitHub.

jaredjj3 committed:
Prevent octave from being -0 when calculating key properties

https://github.com/0xfe/vexflow/issues/1595

rvilarl committed:
Merge pull request #1596 from jaredjj3/fix/tables

Prevent octave from being -0 when calculating key properties

RyoSusami committed:
Fix ornament delayed draw, such as on a grand staff.

RyoSusami committed:
Fix ornament delayed draw, consider the modifier width at the beginning of the stave.

RyoSusami committed:
Added test cases for Ornament, for testing delayed turns.

RyoSusami committed:
Fix ornament delayed draw, improved selection of nextContext.

RyoSusami committed:
Delete unnecessary imports from Ornament.

rvilarl committed:
Merge pull request #1598 from RyoSusami/hotfix/ornament_delayed_draw

Fix ornament delayed draw, such as on a grand staff.

AaronDavidNewman committed:
fix bug 1603 - key signature on non-treble stave, no glyph

None of our test cases had clef with non-default clef (e.g. bass) and a key signature, without the clef symbol displayed.  I added a test case and created a 'addClefLines' that adds the logical clef (e.g. the lines match the clef) but without the glyph.

AaronDavidNewman committed:
fix the note in the test case to be middle C

AaronDavidNewman committed:
code review comments

AaronDavidNewman committed:
Merge pull request #1606 from AaronDavidNewman/master

issue #1603 - key signature on non-treble stave, no glyph

judocode committed:
open context for barnote

judocode committed:
assert getSVGElement is not undefined

judocode committed:
only check if getSVGElement is undefined if it's the SVG renderer

AaronDavidNewman committed:
Merge pull request #1618 from judocode/master

fix: BarNote getSVGElement always returning undefined

ronyeh committed:
Release VexFlow 4.2.4

ronyeh committed:
Remove build/ after releasing version 4.2.4 to npm and GitHub.

ronyeh committed:
VexFlow 4.x will be accessible as 'vexflow4' on npm.
This allows us to continue to patch 4.x.y while maintaining the current major version as 'vexflow' on npm.

ronyeh committed:
Update package-lock.json

ronyeh committed:
Release VexFlow 4.2.5

ronyeh committed:
Remove build/ after releasing version 4.2.5 to npm and GitHub.

jeremy9959 committed:
added 3 additional standard banjo tunings: doubleCBanjo, doubleDBanjo, sawmillBanjo

jeremy9959 committed:
Update tuning.ts

I removed the extra  spaces.

ronyeh committed:
Merge pull request #1623 from jeremy9959/banjo_tunings

added 3 additional standard banjo tunings: doubleCBanjo, doubleDBanjo…
